sample_project/Assets/SampleViewer/Samples/BuildingFilter/BuildingFilter.cs

```diff
@@ -0,0 +1,502 @@
+                }
+            }
+            else if (levelNumber > 0)
+            {
+                --levelNumber;
+                levelText.SetText(levelNumber.ToString());
+                if (int.TryParse(phaseText.text, out phaseNumber))
+                {
+                    GenerateWhereClause(levelNumber, phaseNumber, false, false);
+                }
+            }
+        });
+
+        rightLevel.onClick.AddListener(delegate
+        {
+            if (!int.TryParse(levelText.text, out levelNumber))
+            {
+                levelNumber = buildingStatistics.bldgLevelMax;
+                levelText.SetText(levelNumber.ToString());
+                if (int.TryParse(phaseText.text, out phaseNumber))
+                {
+                    GenerateWhereClause(levelNumber, phaseNumber, false, false);
+                }
+            }
+
+            if (levelNumber < buildingStatistics.bldgLevelMax)
+            {
+                ++levelNumber;
+                levelText.SetText(levelNumber.ToString());
+                if (int.TryParse(phaseText.text, out phaseNumber))
+                {
+                    GenerateWhereClause(levelNumber, phaseNumber, false, false);
+                }
+            }
+        });
+
+        phaseSlider.onValueChanged.AddListener(delegate (float value)
+        {
+            int phaseRange = buildingStatistics.createdPhaseMax - buildingStatistics.createdPhaseMin;
+            int phase = Mathf.RoundToInt(value * phaseRange) + buildingStatistics.createdPhaseMin;
+
+            if (int.TryParse(levelText.text, out levelNumber))
+            {
+                GenerateWhereClause(levelNumber, phase, false, false);
+            }
+            else
+            {
+                GenerateWhereClause(buildingStatistics.bldgLevelMax, phase, false, true);
+            }
+
+            if (int.TryParse(phaseText.text, out phaseNumber))
+            {
+                phaseText.SetText(phase.ToString());
+            }
+        });
+
+        leftPhase.onClick.AddListener(delegate
+        {
+            phaseSlider.value = Mathf.Clamp(phaseSlider.value - buildingStatistics.stepSize, 0f, 1f);
+        });
+
+        rightPhase.onClick.AddListener(delegate
+        {
+            phaseSlider.value = Mathf.Clamp(phaseSlider.value + buildingStatistics.stepSize, 0f, 1f);
+        });
+    }
+
+    // Update is called once per frame
+    private void Update()
+    {
+    }
+
+    private void InitializeBuildingSceneLayer()
+    {
+        arcGISMapComponent = FindObjectOfType<ArcGISMapComponent>();
+        if (arcGISMapComponent != null)
+        {
+            var allLayers = arcGISMapComponent.Map.Layers;
+            for (ulong i = 0; i < allLayers.GetSize(); i++)
+            {
+                if (allLayers.At(i).GetType().Name == "ArcGISBuildingSceneLayer")
+                {
+                    buildingSceneLayer = allLayers.At(i) as ArcGISBuildingSceneLayer;
+                }
```

Comment:
Are there ever cases where we have multiple BSLs in the layers? Should we break after finding the first one?

Reply:
I don't think it would matter because there's only 1 layer added but just in case I added a break and confirmed it works the same.